### PR TITLE
Feature/40385

### PR DESCRIPTION
**Pull request by UrvashiChaudhary**
Task:
1. Install react using following commands:
* npx create-react-app training
* mv training/.* .
* rm -rf training
2. Remove all unwanted files i.e. App.css, logo.svg, index.css
3. Remove irrelevant code from index.js and App.js, and from render method of “App.js”
4. Rename the file App.js to App.jsx, index.js to index.jsx and App.test.js to App.test.jsx
4. Setup following tools:
* editorconfig - Will be used by IDE for indentation purpose.
* linting - Must run with command *npm run lint*, please do install *eslint-config-airbnb* as dev-dependencies.
* husky - “Add lint and test as pre commit and push hooks”, please do install *husky* as dev-dependencies.
* commitzen - must run via command *npm run commit*, and from today onwards use this command to commit your code.
Please find the attachment for editorconfig, linting, and huskyrc configurations.
Make sure “EditorConfig for VS Code” and “ESLint“ Plugins must be installed in VScode.
5. Make sure below commands are working:
* *npm run test*, - Will run existing test cases
* *npm run test:c*, - Will run existing test cases with coverage
* *npm run lint*, - Linting of “src” folder.
* *npm run lint:fix*, - Fix of all lint related problems inside “src” folder.
* *npm run commit* - Must prompt user during during commit.
6. Disable the linting of the serviceWorker.js file using */* eslint-disable */* at the top of the page.
7. Make sure all the lint errors have been fixed.
8. Now create following mentioned filed as shown in the tree structure: https://gyazo.com/f9c7e38c399ddd441fbd050c70cc159a
9. TextField is a component that will render a text input field, and will accept any of the following three props:
* value - string (value will be rendered inside text field)
* error - string (Show the red border with an error message)
* disabled - boolean (disable the text field)
10. TextFieldDemo will be used to print the different flavors of the TextField Component as shown here: https://gyazo.com/299394da8400675e4b036e047cf4e322
11. TextFieldDemo will be rendered inside the App.jsx field.
13. style.js will contain the styling of TextField
14. Index.js file place at the root of components and pages, will be used to export the various components.

### CODE REVIEW

File: training/src/components/TextField/TextField.jsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+
+import PropTypes from 'prop-types';
+
+import { Error, Input } from './style';
+
+const TextField = (props) => {
+  const { value, disabled, error } = props;
+  if (error) {
```
Comment (adilsuccessive) [inline training/src/components/TextField/TextField.jsx]: Optimise the code
If-else block is not necessary

File: training/src/pages/TextFieldDemo/TextFieldDemo.jsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+import { TextField } from '../../components';
+
+import { Div } from '../../components/TextField/style';
+
+const TextFieldDemo = () => (
+  <Div>
+    <p><b>This is a Disabled Input</b></p>
+    <TextField disabled value="disabled input" />
+    <p><b> A Valid Inpu</b></p>
```
Comment (adilsuccessive) [inline training/src/pages/TextFieldDemo/TextFieldDemo.jsx]: spelling mistake (A Valid Inpu)

File: training/src/App.jsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+// import { render } from '@testing-library/react';
+import React from 'react';
+import { TextFieldDemo } from './pages';
+
+function App() {
```
Comment (adilsuccessive) [inline training/src/App.jsx]: make this arrow function

**Review by adilsuccessive — CHANGES_REQUESTED** [review]
remove package-lock.json from pr

File: training/src/components/TextField/TextField.jsx (hunk 1 above)
Comment (UrvashiChaudhary) [reply training/src/components/TextField/TextField.jsx]: done

**Comment by UrvashiChaudhary** [comment]
> remove package-lock.json from pr

removed

File: training/src/pages/TextFieldDemo/TextFieldDemo.jsx (hunk 2 above)
Comment (UrvashiChaudhary) [reply training/src/pages/TextFieldDemo/TextFieldDemo.jsx]: resolved

File: training/src/App.jsx (hunk 3 above)
Comment (UrvashiChaudhary) [reply training/src/App.jsx]: done

File: training/src/components/TextField/TextField.jsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+
+import PropTypes from 'prop-types';
+
+import { Error, Input } from './style';
+
+const TextField = (props) => {
```
Comment (adilsuccessive) [inline training/src/components/TextField/TextField.jsx:7]: disabled prop is missing.
all the 3 text fields are of error variant on rendering.

**Review by adilsuccessive — CHANGES_REQUESTED** [review]

File: training/src/components/TextField/TextField.jsx (hunk 4 above)
Comment (UrvashiChaudhary) [reply training/src/components/TextField/TextField.jsx:7]: done

**Review by adilsuccessive — APPROVED** [review]

**Pull request merged**